spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -1,11 +1,36 @@
 RSpec.describe "expect { ... }.to raise_error" do
   it_behaves_like("an RSpec matcher", :valid_value => lambda { raise "boom" },
                                       :invalid_value => lambda { }) do
-    let(:matcher) { raise_error }
+    let(:matcher) { raise_error Exception }
   end
 
   it "passes if anything is raised" do
-    expect {raise}.to raise_error
+    expect { raise "error" }.to raise_error "error"
+  end
+
+  it "issues a warning when used without an error class or message" do
+    expect_warning_with_call_site __FILE__, __LINE__+1, /without providing a specific error/
+    expect { raise }.to raise_error
+  end
+
+  it "can supresses the warning when configured to do so", :warn_about_false_positives do
```

Comment:
The suppression of the warning via the config is the point of this example, so it seems like it should be front and center in the example rather then hidden in a shared context.  I was expecting the shared context to be used to guarantee the config value is reset back to what it started at, but not to actually make the state change itself.